ext/mbstring/php_mbregex.c

```diff
@@ -1051,28 +1053,25 @@ static void _php_mb_regex_ereg_replace_exec(INTERNAL_FUNCTION_PARAMETERS, OnigOp
 		}
 
 		if (!php_mb_check_encoding(string, string_len, enc)) {
-			RETURN_NULL();
+			zend_argument_value_error(3, "must be a valid string in '%s'", php_mb_regex_get_mbctype());
+			RETURN_THROWS();
 		}
 
 		if (option_str != NULL) {
-			_php_mb_regex_init_options(option_str, option_str_len, &options, &syntax, &eval);
+			/* Initialize option and in case of failure it means there is a value error */
+			if(!_php_mb_regex_init_options(option_str, option_str_len, &options, &syntax, 4)) {
+				RETURN_THROWS();
+			}
 		} else {
 			options |= MBREX(regex_default_options);
 			syntax = MBREX(regex_default_syntax);
 		}
 	}
-	if (eval) {
-		if (is_callable) {
-			php_error_docref(NULL, E_WARNING, "Option 'e' cannot be used with replacement callback");
-		} else {
-			php_error_docref(NULL, E_WARNING, "The 'e' option is no longer supported, use mb_ereg_replace_callback instead");
-		}
-		RETURN_FALSE;
-	}
 
 	/* create regex pattern buffer */
 	re = php_mbregex_compile_pattern(arg_pattern, arg_pattern_len, options, syntax);
 	if (re == NULL) {
+		// Should this be considered an error instead?
```

Comment:
Probably not, as both mbregex and pcre don't provide an independent facility to validate regular expressions, as far as I know.